.github/workflows/ci-e2e.yml

```diff
@@ -100,6 +93,123 @@ jobs:
                   echo "Image size: $(numfmt --to=iec --suffix=B --format="%.2f" $image_size_bytes)" >> $GITHUB_STEP_SUMMARY
                   echo "Image name: $image_name" >> $GITHUB_STEP_SUMMARY
 
+    push:
+        # We use artifacts to pass the docker image between steps, so we do
+        # the push separately such that we can do this in parallel with other
+        # jobs.
+        name: Docker image push
+        if: ${{ github.event_name == 'push' || github.event.pull_request.head.repo.full_name == 'PostHog/posthog' }}
+        runs-on: ubuntu-latest
+        needs: build
+
+        steps:
+            - name: Login to GitHub Container Registry
+              uses: docker/login-action@v1
+              with:
+                  registry: ghcr.io
+                  username: ${{ github.actor }}
+                  password: ${{ secrets.GITHUB_TOKEN }}
+
+            - uses: actions/download-artifact@v3
+              with:
+                  name: docker-image
+                  path: /tmp/
+
+            - name: Push (if PostHog/posthog)
+              # Only push to GHCR if running on PostHog/posthog, as there's no read-write token on forks
+              run: |
+                  docker load < /tmp/posthog-image.tar
+                  docker image push --all-tags ghcr.io/posthog/posthog/posthog
+
+    posthog_cloud:
+        name: PostHog Cloud image build
+        runs-on: ubuntu-latest
+        needs: [build]
+
+        steps:
+            - name: Checkout PR branch
+              uses: actions/checkout@v2
+
+            - uses: actions/download-artifact@v3
+              with:
+                  name: docker-image
+                  path: /tmp/
+
+            - name: Load the non-cloud image
+              id: load
+              run: |
+                  set -x
+                  docker load < /tmp/posthog-image.tar
+                  docker image inspect ${{ needs.build.outputs.docker_image_id }}
+                  echo "::set-output name=docker_image_digest::$(docker image inspect ${{ needs.build.outputs.docker_image_id }}  -f '{{ index .RepoTags 0 }}')"
+
+            - name: Configure AWS credentials
```

Comment:
Nit: we should use OpenID connect (like we do in other repos) instead of hardcoded credentials

Reply:
Oh yeah that looks like a copy and paste. I'l leave this got another PR.